docs/tasks/debug-application-cluster/audit.md

```diff
@@ -91,14 +94,56 @@ audit backend using the following kube-apiserver flags:
 
 - `--audit-webhook-config-file` specifies the path to a file with a webhook
   configuration. Webhook configuration is effectively a [kubeconfig][kubeconfig].
-- `--audit-webhook-mode` define the buffering strategy, one of the following:
-  - `batch` - buffer events and asynchronously send the set of events to the external service
-    This is the default
-  - `blocking` - block API server responses on sending each event to the external service
+- `--audit-webhook-initial-backoff` specifies the amount of time to wait after the first failed
+  request before retrying. From the second request and so on, exponential backoff is used.
 
 The webhook config file uses the kubeconfig format to specify the remote address of
 the service and credentials used to connect to it.
 
+### Batching
+
+Both log and webhook backends support batching. Using webhook as an example, here's the list of
+available flags. To get the same flag for log backend, replace `webhook` with `log` in the flag
+name. By default, batching is enabled in `webhook` and disabled in `log`. Similarly, by default
+throttling is enabled in `webhook` and disabled in `log`.
+
+- `--audit-webhook-mode` define the buffering strategy, one of the following:
+  - `batch` - buffer events and asynchronously process then in batches. This is the default
+  - `blocking` - block API server responses on processing each individual event
+
+The following flags are only used in the `batch` mode.
+
+- `--audit-webhook-batch-buffer-size` defines the size of the buffer to store events before batching.
+  If the rate of incoming events is too high and the buffer is overflown, events are dropped
+- `--audit-webhook-batch-max-size` defines the maximum number of events in one batch
+- `--audit-webhook-batch-max-wait` defines the maximum amount of time to wait before unconditionally
+  batching events in the queue
+- `--audit-webhook-batch-throttle-qps` defines the maximum average number of batches generated
+  per second
+- `--audit-webhook-batch-throttle-burst` defines the maximum number of batches generated per second
+  if the allowed QPS was underutilized previously
+
+#### How to set up parameters
+
+Parameters should be set to accommodate your load on the apiserver.
+
+For example, if you receive 100 requests to the kube-apiserver each second, and each request is
+audited only on `StageResponseStarted` and `StageResponseComplete` stages, you should account for
+~200 audit events being generated each second. Assuming that you have up to 100 events in a batch,
+you should set throttling level at at least 2 QPS. Assuming that the backend can take up to
+5 seconds to write events, you should set the buffer size to hold up to 5 seconds of events, i.e.
+10 batches, i.e. 1000 events.
+
+In most cases however, the default parameters should be sufficient and you don't have to worry about
+setting them manually. You can look at the Prometheus metric `apiserver_audit_error_total` and
```

Comment:
Does `apiserver_audit_error_total` include dropped events? If it doesn't can we include the dropped metric here too?

Also can you link to a doc explaining how an admin can scrape the API server?